src/openms/source/CHEMISTRY/Element.cpp

```diff
@@ -111,10 +114,34 @@ namespace OpenMS
 
   void Element::setIsotopeDistribution(const IsotopeDistribution & distribution)
   {
-    isotopes_ = distribution;
+    isotope_distr_ = distribution;
+    // TODO: remove this fxn
   }
 
   const IsotopeDistribution & Element::getIsotopeDistribution() const
+  {
+    return isotope_distr_;
+  }
+
+  void Element::setIsotopes(const std::vector<const Isotope*>& isotopes)
+  {
+    isotopes_ = isotopes;
+    updateIsotopeDistr(); // calculate new cached distribution
+  }
+
+  void Element::updateIsotopeDistr()
```

Comment:
can this method be private/protected?

Reply:
It seems to the used in `addIsotope`, but is this part really that performance critical that we burden the user with taking care of updating the isotope distribution if an isotope was replaced?

Reply:
yes `addIsotope` uses it, I guess you could trigger it with `setIsotopes(getIsotopes)` to achieve this?